Runtime/src/main/java/com/rakuten/tech/mobile/perf/runtime/internal/RuntimeContentProvider.java

```diff
@@ -84,10 +84,13 @@ private Config createConfig(@NonNull Context context, @Nullable ConfigurationRes
     }
     Config config = null; // configuration for TrackingManager
 
-    double enablePercent = lastConfig.getEnablePercent();
-
-    double randomNumber = new Random(System.currentTimeMillis()).nextDouble() * 100.0;
-    if (randomNumber <= enablePercent) {
+    boolean enable = Util.isAppDebuggable(context);
+    if (enable == false) {
+      double enablePercent = lastConfig.getEnablePercent();
+      double randomNumber = new Random(System.currentTimeMillis()).nextDouble() * 100.0;
+      enable = (randomNumber <= enablePercent);
+    }
+    if (enable) {
```

Comment:
If the app is debuggable, then you don't need to generate the random number. I will try to make code more readable.

Reply:
yes that's true, but performance is not the bottle neck here. let's focus on clarity - it could also be an `if then else` block (more verbose, no performance overhead)